sim/shaman/runes.go

```diff
@@ -95,18 +121,59 @@ func (shaman *Shaman) applyMaelstromWeapon() {
 	})
 }
 
+const ShamanPowerSurgeProcChance = .05
+
 func (shaman *Shaman) applyPowerSurge() {
 	if !shaman.HasRune(proto.ShamanRune_RuneWaistPowerSurge) {
 		return
 	}
 
+	var affectedSpells []*core.Spell
+
+	procAura := shaman.RegisterAura(core.Aura{
+		Label:    "Power Surge Proc",
+		ActionID: core.ActionID{SpellID: 440285},
+		Duration: time.Second * 10,
+		OnInit: func(aura *core.Aura, sim *core.Simulation) {
+			affectedSpells = core.FilterSlice(
+				core.Flatten([][]*core.Spell{
+					shaman.ChainLightning,
+					shaman.LightningBolt,
+					{shaman.LavaBurst},
+				}), func(spell *core.Spell) bool { return spell != nil },
+			)
+		},
+		OnGain: func(aura *core.Aura, sim *core.Simulation) {
+			core.Each(affectedSpells, func(spell *core.Spell) { spell.CastTimeMultiplier -= 1 })
+		},
```

Comment:
Also in the `OnGain` we would want to reset the cooldowns of Lava Burst and Chain Lightning